### PR TITLE
ci: add detailed perf parquet diffs

### PR DESCRIPTION
**Pull request by jasonqinzhou**
## Summary

- generate row-level perf parquet diffs for added, removed, and modified rows
- upload the full review bundle: `comment.md`, `parquet-diff-details/changed-files.csv`, `parquet-diff-details/summary.csv`, exact row CSVs, and full per-file unified diffs under `parquet-diff-details/diffs/`
- keep the PR comment compact with totals, inline row previews, and links into the artifact bundle instead of duplicating the normal PR file diff
- support manual workflow dispatch with explicit base/head refs so the report workflow can be smoke-tested against another PR, such as PR #1114

## Notes

The row matcher treats non-measurement columns as the logical perf key and reports measurement changes as modified rows. When a table has duplicate logical keys, unmatched rows are paired deterministically within each key so repeated perf samples show up as modified rows instead of noisy whole-table remove/add churn.

The report no longer includes the old `Other Parquet Changes` section. Full per-file diffs now live in the uploaded artifact, where reviewers can inspect every changed perf data file without bloating the PR comment.

## Validation

- `uv run pytest tests/unit/tools/test_parquet_diff.py`
- `uv run ruff check .`
- `uv run ruff format --check .`
- `git diff --check`
- CI on `95c396db`: CodeRabbit, DCO, parquet-diff, Ruff, ESLint, copyright, unit, e2e, Cargo Deny, and PR title checks passed; accuracy regression was still running when this description was updated.
- Smoke-tested the generated report against PR #1114 with the manual workflow; artifact bundle: https://github.com/ai-dynamo/aiconfigurator/actions/runs/26847997281/artifacts/7369840583

## Summary by CodeRabbit

* **New Features**
  * Row-level performance diffs with per-file CSV artifacts, truncated inline CSV previews, and aggregated added/removed/modified row totals.

* **CI**
  * Workflows support manual dispatch, fetch base/head refs, and upload both the summary report and detailed row-diff artifacts; PR comments now include links to the artifact bundle.

* **Tests**
  * New unit tests validating row-level diffs, duplicate-key pairing, artifact generation, and report/CSV previews.

* **Documentation**
  * PR-review docs updated with new report arguments and artifact expectations.

### CODE REVIEW

**Comment by jasonqinzhou** [comment]
## Example smoke-test output

I smoke-tested this branch's enhanced `tools/perf_database/parquet_diff.py` against PR #1114 to show what reviewers will see on a perf-data PR.

Command used locally:

```bash
uv run python tools/perf_database/parquet_diff.py \
  --base-ref origin/main \
  --head-ref tmp-pr-1114-detail-test \
  --output /tmp/aic-parquet-comment.md \
  --detail-dir /tmp/aic-parquet-details
```

The run produced `36` exact row-level CSV files under `/tmp/aic-parquet-details/`, including `summary.csv`. In CI, those files are uploaded in the `perf-parquet-diff` artifact under `parquet-diff-details/`.

<details open>
<summary>Example Perf Parquet Diff Report</summary>

## Perf Parquet Diff Report

Compared `origin/main` to `tmp-pr-1114-detail-test` for `src/aiconfigurator/systems/data`.

- Parquet files changed: 28
- CSV-to-parquet conversions checked: 0
- Conversions with matching columns and rows: 0
- New parquet files without a base CSV/parquet counterpart: 1
- Modified or renamed parquet files: 27
- Deleted parquet files: 0
- Legacy `*_perf.txt` files added or modified: 0
- Row-level changes: +82743 / -0 / ~130723

### Other Parquet Changes

| Status | File | Base rows | Head rows | Rows + | Rows - | Rows ~ | Details |
| --- | --- | --- | --- | --- | --- | --- | --- |
| A | src/aiconfigurator/systems/data/rtx_pro_6000_server/nccl/2.28.9/nccl_perf.parquet | None | 168 | 168 | 0 | 0 | added |
| M | src/aiconfigurator/systems/data/b200_sxm/sglang/0.5.10/context_attention_perf.parquet | 33714 | 35224 | 1510 | 0 | 33714 | added, modified; duplicate keys; unmatched rows paired within key |
| M | src/aiconfigurator/systems/data/b200_sxm/sglang/0.5.10/dsv4_csa_context_module_perf.parquet | 752 | 765 | 13 | 0 | 0 | added |
| M | src/aiconfigurator/systems/data/b200_sxm/sglang/0.5.10/dsv4_csa_generation_module_perf.parquet | 1600 | 1616 | 16 | 0 | 0 | added |
| M | src/aiconfigurator/systems/data/b200_sxm/sglang/0.5.10/dsv4_hca_attn_module_perf.parquet | 1656 | 1688 | 32 | 0 | 0 | added |
| M | src/aiconfigurator/systems/data/b200_sxm/sglang/0.5.10/dsv4_hca_context_module_perf.parquet | 752 | 765 | 13 | 0 | 0 | added |
| M | src/aiconfigurator/systems/data/b200_sxm/sglang/0.5.10/dsv4_hca_generation_module_perf.parquet | 1600 | 1616 | 16 | 0 | 0 | added |
| M | src/aiconfigurator/systems/data/b200_sxm/sglang/0.5.10/dsv4_paged_mqa_logits_module_perf.parquet | 1937 | 1945 | 8 | 0 | 0 | added |
| M | src/aiconfigurator/systems/data/b200_sxm/sglang/0.5.10/gemm_perf.parquet | 130536 | 131720 | 1184 | 0 | 0 | added |
| M | src/aiconfigurator/systems/data/b200_sxm/sglang/0.5.10/generation_attention_perf.parquet | 19484 | 21066 | 1582 | 0 | 19484 | added, modified; duplicate keys; unmatched rows paired within key |
| M | src/aiconfigurator/systems/data/b200_sxm/sglang/0.5.10/mhc_module_perf.parquet | 70 | 140 | 70 | 0 | 0 | added |
| M | src/aiconfigurator/systems/data/b200_sxm/sglang/0.5.10/moe_perf.parquet | 116940 | 117444 | 504 | 0 | 0 | added; duplicate keys; unmatched rows paired within key |
| M | src/aiconfigurator/systems/data/b200_sxm/trtllm/1.3.0rc10/context_attention_perf.parquet | 63123 | 63244 | 121 | 0 | 0 | added; duplicate keys; unmatched rows paired within key |
| M | src/aiconfigurator/systems/data/b200_sxm/trtllm/1.3.0rc10/dsa_context_module_perf.parquet | 3915 | 3957 | 42 | 0 | 0 | added |
| M | src/aiconfigurator/systems/data/b200_sxm/trtllm/1.3.0rc10/dsa_generation_module_perf.parquet | 8824 | 8856 | 32 | 0 | 0 | added |
| M | src/aiconfigurator/systems/data/b200_sxm/trtllm/1.3.0rc10/gemm_perf.parquet | 130299 | 131483 | 1184 | 0 | 0 | added |
| M | src/aiconfigurator/systems/data/b200_sxm/trtllm/1.3.0rc10/generation_attention_perf.parquet | 40240 | 40565 | 325 | 0 | 0 | added |
| M | src/aiconfigurator/systems/data/b200_sxm/trtllm/1.3.0rc10/moe_perf.parquet | 105543 | 106684 | 1141 | 0 | 77 | added, modified |
| M | src/aiconfigurator/systems/data/b200_sxm/vllm/0.19.0/context_attention_perf.parquet | 21816 | 22086 | 270 | 0 | 21816 | added, modified; duplicate keys; unmatched rows paired within key |
| M | src/aiconfigurator/systems/data/b200_sxm/vllm/0.19.0/gemm_perf.parquet | 114566 | 115750 | 1184 | 0 | 0 | added |
| M | src/aiconfigurator/systems/data/b200_sxm/vllm/0.19.0/generation_attention_perf.parquet | 16908 | 17249 | 341 | 0 | 16908 | added, modified; duplicate keys; unmatched rows paired within key |
| M | src/aiconfigurator/systems/data/b200_sxm/vllm/0.19.0/moe_perf.parquet | 56781 | 58139 | 1358 | 0 | 0 | added; duplicate keys; unmatched rows paired within key |
| M | src/aiconfigurator/systems/data/h200_sxm/sglang/0.5.10/gemm_perf.parquet | 101010 | 136752 | 35742 | 0 | 0 | added |
| M | src/aiconfigurator/systems/data/h200_sxm/vllm/0.19.0/gemm_perf.parquet | 101004 | 136746 | 35742 | 0 | 0 | added |
| M | src/aiconfigurator/systems/data/rtx_pro_6000_server/sglang/0.5.10/moe_perf.parquet | 78395 | 78399 | 4 | 0 | 0 | added; duplicate keys; unmatched rows paired within key |
| M | src/aiconfigurator/systems/data/rtx_pro_6000_server/vllm/0.19.0/context_attention_perf.parquet | 21816 | 21848 | 32 | 0 | 21816 | added, modified; duplicate keys; unmatched rows paired within key |
| M | src/aiconfigurator/systems/data/rtx_pro_6000_server/vllm/0.19.0/generation_attention_perf.parquet | 16908 | 16988 | 80 | 0 | 16908 | added, modified; duplicate keys; unmatched rows paired within key |
| M | src/aiconfigurator/systems/data/rtx_pro_6000_server/vllm/0.19.0/moe_perf.parquet | 46494 | 46523 | 29 | 0 | 0 | added; duplicate keys; unmatched rows paired within key |

Exact row-level CSVs are attached to the workflow artifact under `parquet-diff-details/` with filenames matching `<source>.{added,removed,modified}.csv`.

</details>

**Comment by jasonqinzhou** [comment]
> ## Perf Parquet Diff Report
> * Artifact bundle: [perf-parquet-diff](https://github.com/ai-dynamo/aiconfigurator/actions/runs/26847997281/artifacts/7369840583) (retained for 30 days)
> 
> Compared `origin/main` to `HEAD` for `src/aiconfigurator/systems/data`.
> 
> * Parquet files changed: 21
> * CSV-to-parquet conversions checked: 0
> * Conversions with matching columns and rows: 0
> * New parquet files without a base CSV/parquet counterpart: 0
> * Modified or renamed parquet files: 21
> * Deleted parquet files: 0
> * Legacy `*_perf.txt` files added or modified: 0
> * Row-level changes: +10946 / -0 / ~91999
> * Full per-file diff artifacts: 21 files under `parquet-diff-details/diffs/`
> 
> ### Artifact Contents
> * Full per-file unified diffs: [perf-parquet-diff](https://github.com/ai-dynamo/aiconfigurator/actions/runs/26847997281/artifacts/7369840583) artifact under `parquet-diff-details/diffs/` (21 files)
> * Exact row-level CSVs: [perf-parquet-diff](https://github.com/ai-dynamo/aiconfigurator/actions/runs/26847997281/artifacts/7369840583) artifact under `parquet-diff-details/` (listed in `summary.csv`)

Example report for [PR1114](https://github.com/ai-dynamo/aiconfigurator/pull/1114)

**Comment by jasonqinzhou** [comment]
@coderabbitai resolve

**Review by Harrilee — APPROVED** [review]
